src/main/java/org/osiam/ResourceServer.java

```diff
@@ -0,0 +1,220 @@
+    public DataSource dataSource() {
+        return new HikariDataSource(hikariConfig());
+    }
+
+    @Bean(initMethod = "migrate")
+    @Autowired
+    public Flyway flyway(DataSource dataSource,
+            @Value("${org.osiam.resource-server.db.vendor:postgresql}") String databaseVendor) {
+        Flyway flyway = new Flyway();
+        flyway.setBaselineOnMigrate(true);
+        flyway.setDataSource(dataSource);
+        flyway.setLocations("db/migration/" + databaseVendor);
+        flyway.setTable("resource_server_schema_version");
+        MigrationVersion version = MigrationVersion.fromVersion("0");
+        flyway.setBaselineVersion(version);
+        return flyway;
+    }
+
+    @Bean
+    public HikariConfig hikariConfig() {
+        HikariConfig hikariConfig = new HikariConfig();
+        hikariConfig.setPoolName("osiam-auth-server-cp");
+        hikariConfig.setDriverClassName(driverClassName);
+        hikariConfig.setJdbcUrl(databaseUrl);
+        hikariConfig.setUsername(databaseUserName);
+        hikariConfig.setPassword(databasePassword);
+        return hikariConfig;
+    }
+
+    @Bean
+    @DependsOn("flyway")
+    public EntityManagerFactory entityManagerFactory() throws SQLException {
+
+        HibernateJpaVendorAdapter vendorAdapter = new HibernateJpaVendorAdapter();
+        vendorAdapter.setGenerateDdl(false);
+        vendorAdapter.setShowSql(false);
+        vendorAdapter.setDatabasePlatform(databasePlatform);
+
+        Properties jpaProperties = new Properties();
+        jpaProperties.put("hibernate.hbm2ddl.auto", "validate");
+        jpaProperties.put("hibernate.ejb.naming_strategy", "org.hibernate.cfg.ImprovedNamingStrategy");
+
+        LocalContainerEntityManagerFactoryBean factory = new LocalContainerEntityManagerFactoryBean();
+        factory.setJpaVendorAdapter(vendorAdapter);
+        factory.setPackagesToScan("org.osiam.storage.entities");
+        factory.setDataSource(dataSource());
+        factory.setJpaProperties(jpaProperties);
+        factory.afterPropertiesSet();
+
+        return factory.getNativeEntityManagerFactory();
+    }
+
+    @Bean
+    public JpaTransactionManager transactionManager() throws SQLException {
+        JpaTransactionManager txManager = new JpaTransactionManager();
+        txManager.setEntityManagerFactory(entityManagerFactory());
+        return txManager;
+    }
+
+    @Bean
+    public MetricRegistry metricRegistry() {
+        return new MetricRegistry();
+    }
+
+    @Bean
+    public HibernateExceptionTranslator hibernateExceptionTranslator() {
+        return new HibernateExceptionTranslator();
+    }
+
+    @Configuration
+    @EnableResourceServer
+    protected static class ResourceServerConfiguration extends ResourceServerConfigurerAdapter {
+
+        @Autowired
+        private AccessTokenValidationService accessTokenValidationService;
+
+        @Bean
+        public OAuth2AuthenticationEntryPoint createEntryPoint() {
```

Comment:
yes, and the realm should be `oauth2-resource-server` instead of `oauth2-authorization-server`